sympy/utilities/lambdify.py

```diff
@@ -297,7 +297,18 @@ def lambdify(args, expr, modules=None, printer=None, use_imps=True,
     """
     from sympy.core.symbol import Symbol
     from sympy.utilities.iterables import flatten
-
+    invalid_str_expr = 0
+    invalid_str_args = 0
+    if isinstance(expr, str):
+        if not tuple(expr.replace(',','')) == tuple(expr.split(',')):
+            invalid_str_expr = 1
+    if isinstance(args, str):
+        if not tuple(args.replace(',','')) == tuple(args.split(',')):
+            invalid_str_args = 1
```

Comment:
You can do the ```raise``` statement right here.
In that case, ```invalid_str_args``` isn't necessary anymore (and my proposal to use booleans instead of integers becomes a non-issue).